### ISSUE TITLE
Android ANR in Broadcast of Intent { act=android.provider.Telephony.SMS_RECEIVED flg=0x8000010 ... }

### ISSUE THREAD

**Issue by mitonline** [OneSignal/OneSignal-Android-SDK#237]
We are seeing a lot of ANR's with onesignal on the developer console. Please see attached stacktrace.
We are using the latest onesignal android native sdk.  Please help.
From build.gradle
    compile 'com.onesignal:OneSignal:[3.5.3,4.0.0)'

[onesignal_anr.txt](https://github.com/OneSignal/OneSignal-Android-SDK/files/1034895/onesignal_anr.txt)

**Comment by jkasten2** [comment]
@mitonline Thanks for the full log, we have identified the main thread locking case from it. We will include a fix in the next release for this.

**Comment by mitonline** [comment]
@jkasten2 Thanks Josh. We are also seeing a similar ANR but for a different broadcast intent. Have raised a new issue #239 for that one for reference. Please mark it duplicate if both of them have the same root cause.
Any rough estimate of timelines that you can give on when the next release will be available for us to update will be helpful ?
Thank you.

**Comment by jkasten2** [comment]
The root of the ANR is from locking the main thread noted by this stack trace from your log.

```Java
"main" prio=5 tid=1 TimedWaiting
  | group="main" sCount=1 dsCount=0 obj=0x76628fa0 self=0x55a5c29410
  | sysTid=20889 nice=-1 cgrp=bg_non_interactive sched=0/0 handle=0x7f94d18fc8
  | state=S schedstat=( 134303436 18258804 133 ) utm=10 stm=3 core=0 HZ=100
  | stack=0x7febbc3000-0x7febbc5000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x04a1d6a8> (a java.lang.Object)
  at java.lang.Thread.join (Thread.java:786)
- locked <0x04a1d6a8> (a java.lang.Object)
  at java.lang.Thread.join (Thread.java:744)
  at com.onesignal.OneSignalRestClient.makeRequest (OneSignalRestClient.java:102)
  at com.onesignal.OneSignalRestClient.putSync (OneSignalRestClient.java:82)
  at com.onesignal.OneSignalStateSynchronizer.syncUserState (OneSignalStateSynchronizer.java:586)
  at com.onesignal.SyncService$2$1.complete (SyncService.java:91)
  at com.onesignal.LocationGMS.fireComplete (LocationGMS.java:208)
- locked <0x063fc5c1> (a java.lang.Class<com.onesignal.i>)
  at com.onesignal.LocationGMS.receivedLocationPoint (LocationGMS.java:233)
  at com.onesignal.LocationGMS.access$200 (LocationGMS.java:54)
  at com.onesignal.LocationGMS.access$302 (LocationGMS.java:54)
  at com.onesignal.LocationGMS$GoogleApiClientListener.onConnected (LocationGMS.java:247)
  at com.google.android.gms.common.internal.zzm.zzq (zzm.java)
- locked <0x0bae3166> (a java.lang.Object)
  at com.google.android.gms.internal.zzaal.zzo (zzaal.java)
  at com.google.android.gms.internal.zzaaj.zzvE (zzaaj.java)
  at com.google.android.gms.internal.zzaaj.onConnected (zzaaj.java)
  at com.google.android.gms.internal.zzaan.onConnected (zzaan.java)
  at com.google.android.gms.internal.zzzy.onConnected (zzzy.java)
  at com.google.android.gms.common.internal.zzl$1.onConnected (zzl.java)
  at com.google.android.gms.common.internal.zzf$zzj.zzwZ (zzf.java)
  at com.google.android.gms.common.internal.zzf$zza.zzc (zzf.java)
  at com.google.android.gms.common.internal.zzf$zza.zzu (zzf.java)
  at com.google.android.gms.common.internal.zzf$zze.zzxa (zzf.java)
  at com.google.android.gms.common.internal.zzf$zzd.handleMessage (zzf.java)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:148)
  at android.app.ActivityThread.main (ActivityThread.java:5438)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:738)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:628)
```

Activity looking into this issue today.

**Comment by jkasten2** [comment]
@mitonline This is now fixed, it will be included in our release next week.
